strum_macros/src/macros/strings/from_string.rs

```diff
@@ -56,16 +56,34 @@ pub fn from_string_inner(ast: &DeriveInput) -> syn::Result<TokenStream> {
         let params = match &variant.fields {
             Fields::Unit => quote! {},
             Fields::Unnamed(fields) => {
-                let defaults =
-                    ::core::iter::repeat(quote!(Default::default())).take(fields.unnamed.len());
-                quote! { (#(#defaults),*) }
+                if let Some(ref value) = variant_properties.default_with {
```

Comment:
I wasn't if there was a new-type variant, if we wanted to be able to default each unnamed field.  It'll be easy enough to add this feature/tests

Reply:
If it's not too difficult, let's try to match the conventions that serde uses according to this doc: 

https://serde.rs/variant-attrs.html#deserialize_with


> `#[serde(deserialize_with = "path")]`
>
> Deserialize this variant using a function that is different from its implementation of Deserialize. The given function must be callable as `fn<'de, D>(D) -> Result<FIELDS, D::Error> where D: Deserializer<'de>`, although it may also be generic over the elements of FIELDS. Variants used with deserialize_with are not required be able to derive Deserialize.
> 
> FIELDS is a tuple of all fields of the variant. A unit variant will have () as its FIELDS type.

I'm not too worried about the generic bounds parts, however, if we can return a tuple, that's probably the clearest way to do it. Thoughts?

Reply:
Let me see what I can do!